src/opam.mli

```diff
@@ -21,32 +44,61 @@ module Package : sig
 
   val synopsis : t -> string option
 
-  val of_path : Path.t -> t option Promise.t
-
-  val sexp_of_t : t -> Sexplib.Sexp.t
-
-  val t_of_sexp : Sexplib.Sexp.t -> t
+  val depends : t -> string list option
 end
 
 type t
 
 val make : unit -> t option Promise.t
 
+(** {4 Working with switches} *)
+
+(** Path of the switch on the filesystem.
+
+    If the switch is a local switch, the path is [switch_path ^ _opam],
+    otherwise it is the result of [opam var root]. *)
+val switch_path : t -> Switch.t -> (Path.t, string) result Promise.t
+
+(** List the opam switches available on the system. *)
 val switch_list : t -> Switch.t list Promise.t
 
+(** Checks that the given switch exists. *)
+val switch_exists : t -> Switch.t -> bool Promise.t
+
+(** Return the current opam switch. *)
 val switch_show : ?cwd:Path.t -> t -> Switch.t option Promise.t
 
-val exec : t -> switch:Switch.t -> args:string list -> Cmd.t
+(** Return the compiler version used by the given switch. *)
+val switch_compiler : t -> Switch.t -> string option Promise.t
 
-val exists : t -> switch:Switch.t -> bool Promise.t
+(** Remove the given switch by calling [opam switch remove]. *)
+val switch_remove : t -> Switch.t -> Cmd.t
 
-val equal : t -> t -> bool
+(** {4 Working with packages} *)
 
-val get_switch_packages :
-  t -> Switch.t -> (Package.t list, string) result Promise.t
+(** Return the list of installed packages in the given opam switch. *)
+val packages : t -> Switch.t -> (Package.t list, string) result Promise.t
 
-val get_switch_compiler : t -> Switch.t -> string option Promise.t
+(** Return the list of root packages in the given opam switch.
 
-val remove_switch : t -> Switch.t -> Cmd.t
+    Root packages are packages that have been installed by the user, as opposed
+    to packages that have been installed as a dependency of another package. *)
+val root_packages : t -> Switch.t -> (Package.t list, string) result Promise.t
 
-val uninstall_package : t -> switch:Switch.t -> package:Package.t -> Cmd.t
+(** Return the list of dependencies of the given package in the given switch.
+
+    Return an error if the package does not exist or its dependencies cannot be
+    read. *)
+val package_dependencies :
+  Package.t -> (Package.t list, string) result Promise.t
+
+(** Uninstall a package from a switch by calling [opam uninstall]. *)
+val package_uninstall : t -> Switch.t -> Package.t -> Cmd.t
+
+(** {4 General utilities} *)
+
+(** Execute an opam sub-command with in the given switch. *)
+val exec : t -> Switch.t -> string list -> Cmd.t
```

Comment:
We need an `args` label on naked `string list` parameters like this one.

Reply:
Done